London | SDC-Nov-2025 | Ikenna Agulobi | Sprint 4 | Implement shell tools python

ike-agu opened the pull request:
<!--

You must title your PR like this:

Region | Cohort | FirstName LastName | Sprint | Assignment Title

For example,

London | 25-ITP-May | Carol Owen | Sprint 1 | Alarm Clock

Fill in the template below - remove any sections that don't apply.

Complete the self checklist - replace each empty box in the checklist [ ] with a [x].

Add the label "Needs Review" and you will get review.

Respond to volunteer reviews until the volunteer marks it as "Complete".

-->

## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

This PR contains my implementation of cat, ls and wc shell tool commands in python

ike-agu committed:
Add python implementation of catCommand and .gitignore file

ike-agu committed:
Implement ls commands in python

ike-agu committed:
Implement wc command-line tool in python

ike-agu committed:
Fix comments

DaryaShirokova reviewed (commented):
Looks good! Just added a few comments

ike-agu committed:
Fix line numbering to match cat -n and -b behaviour

ike-agu committed:
Refactor cat implementationinto functions for readability

ike-agu commented on implement-shell-tools/cat/cat.py:
```diff
@@ -0,0 +1,50 @@
+import argparse
+# -------------------------------------------
```
Thanks for your review and  suggestions. I’ve refactored my code by splitting it into small functions with descriptive names.

ike-agu committed:
Refactor wc into reusable function and fix line counting to match Unix wc

ike-agu commented on implement-shell-tools/wc/README.md:
```diff
@@ -15,3 +15,20 @@ It must act the same as `wc` would, if run from the directory containing this RE
+python wc.py sample-files/*
+
+* Just lines
+python wc.py -l sample-files/3.txt
```
Hi @DaryaShirokova - thanks for pointing this out. The issue was caused because I used split("\n"), which overcounts when a file ends with a newline. I've now  updated my implementation to use content.count("\n"), and the output now matches wc as expected.

ike-agu commented:
Hi @DaryaShirokova , apologies for circling back to this PR after such a long time. I should have addressed the feedback earlier.
Anyway!, Thank you for your review and feedback.
I’ve  implemented all the requested changes.
When you have a moment, could you please take another look? Thank you!

ike-agu commented on implement-shell-tools/wc/wc.py:
```diff
@@ -0,0 +1,93 @@
+
+
+def main():
+    # --------------------------------------------------------
```
I've refactored my code and removed comments. Thanks for feedback.

DaryaShirokova commented on implement-shell-tools/wc/wc.py:
```diff
@@ -0,0 +1,78 @@
+    if len(args.paths) > 1:
+        no_flags = not args.l and not args.w and not args.c
+
+        if no_flags:
```
I am not sure this is the correct logic here? e.g. 
```
wc -l sample-files/*
  1 sample-files/1.txt
  1 sample-files/2.txt
  5 sample-files/3.txt
  7 total
```
it still prints total with -l?

ike-agu replied:
@DaryaShirokova, Thanks again for reviewing my PR. I’ve updated the logic so the total line  now matches real wc behaviour. Please take a look when you have a moment. Thank you!

DaryaShirokova commented on implement-shell-tools/wc/wc.py:
```diff
@@ -0,0 +1,78 @@
+import re
+
+
+def main():
```
Optional: you could try to organize the code into multiple methods for readability

DaryaShirokova reviewed (commented):
LGTM overall, just one thing is still slightly different

ike-agu committed:
Fix total line behavior to match wc for flagged multi-file input

Print totals when multiple files are provided, even if -l, -w, or -c flags are used.This matches wc real behaviour.
